packages/@lwc/engine-core/src/framework/renderer.ts

```diff
@@ -46,13 +51,7 @@ export interface Renderer<N = HostNode, E = HostElement> {
     isConnected(node: N): boolean;
     insertGlobalStylesheet(content: string): void;
     assertInstanceOfHTMLElement?(elm: any, msg: string): void;
-    defineCustomElement(
-        name: string,
-        constructor: CustomElementConstructor,
-        options?: ElementDefinitionOptions
-    ): void;
-    getCustomElement(name: string): CustomElementConstructor | undefined;
-    HTMLElement: typeof HTMLElement;
+    getUpgradableElement(name: string): UpgradableCustomElementConstructor;
```

Comment:
this API is responsible for returning a thing that can be `new`d, and when so, it should construct a valid element for the environment.